src/controllers/legacycontrollersettings.cpp

```diff
@@ -274,3 +291,143 @@ QWidget* LegacyControllerEnumSetting::buildInputWidget(QWidget* pParent) {
 
     return pComboBox;
 }
+
+LegacyControllerColorSetting::LegacyControllerColorSetting(
+        const QDomElement& element)
+        : AbstractLegacyControllerSetting(element) {
+    m_defaultValue = QColor(element.attribute("default"));
+    m_savedValue = m_defaultValue;
+    m_editedValue = m_defaultValue;
+    reset();
+    save();
+}
+
+bool LegacyControllerColorSetting::match(const QDomElement& element) {
+    return element.hasAttribute("type") &&
+            QString::compare(element.attribute("type"),
+                    "color",
+                    Qt::CaseInsensitive) == 0;
+}
+
+void LegacyControllerColorSetting::parse(const QString& in, bool* ok) {
+    if (ok != nullptr) {
+        *ok = false;
+    }
+    reset();
+    save();
+
+    m_savedValue = QColor(in);
+    if (!m_editedValue.isValid()) {
+        *ok = false;
+        return;
+    }
+    m_editedValue = m_savedValue;
```

Comment:
this doesn't work properly. you didn't check for `nullptr` on the second access to `*ok` and you never actually set it `true`, potentially leaving it uninitialized (which should never be the case for an out pointer like this).